src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -107,6 +110,51 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        [HttpGet]
+        [Route("admin/host/workerProcesses")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> GetWorkerProcesses([FromServices] IScriptHostManager scriptHostManager)
+        {
+            if (!Utility.TryGetHostService(scriptHostManager, out IWebHostRpcWorkerChannelManager webHostLanguageWorkerChannelManager))
+            {
+                return StatusCode(StatusCodes.Status503ServiceUnavailable);
+            }
+
+            var processes = new List<WorkerProcessInfo>();
+            string workerRuntime = _environment.GetFunctionsWorkerRuntime();
+
+            var webhostChannelDictionary = webHostLanguageWorkerChannelManager.GetChannels(workerRuntime);
+
+            List<IRpcWorkerChannel> channels = new List<IRpcWorkerChannel>();
+            if (webhostChannelDictionary is not null)
+            {
+                foreach (var pair in webhostChannelDictionary)
+                {
+                    var workerChannel = await pair.Value.Task;
+                    channels.Add(workerChannel);
+                }
+            }
+
+            if (Utility.TryGetHostService(scriptHostManager, out IJobHostRpcWorkerChannelManager jobHostLanguageWorkerChannelManager))
+            {
+                channels.AddRange(jobHostLanguageWorkerChannelManager.GetChannels(workerRuntime));
```

Comment:
Would there be any concern at all if we **don't** get `jobHostLanguageWorkerChannelManager` and the channels from it? Is that normal?

Reply:
It should be okay. In some cases the jobHostLanguageWorkerChannelManager may not have channels any as they may have initialized at webhost level. Also, in case script host is restarting when the API call is made it is possible that we may not get jobHostLanguageWorkerChannelManager.